testutils/junithelper/main.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"io/ioutil"
+	"log"
+
+	"knative.dev/pkg/test/junit"
+)
+
+var (
+	suite  string
+	name   string
+	errMsg string
+	dest   string
+)
+
+func main() {
+	flag.StringVar(&suite, "suite", "", "Name of suite")
+	flag.StringVar(&name, "name", "", "Name of test")
+	flag.StringVar(&errMsg, "err-msg", "", "Error message, empty means test passed, default empty")
+	flag.StringVar(&dest, "dest", "junit_result.xml", "Where junit xml writes to")
+	flag.Parse()
+
+	suites := junit.TestSuites{}
+	suite := junit.TestSuite{Name: suite}
+	var errP *string
```

Comment:
Do we need this? Cannot we just use `&errMsg`?

Reply:
Unfortunately we can't, it has to be null pointer when test passed, but errMsg always got assigned the default value and is never null